ES-5 compatible mappings and related changes

AmitPhulera opened the pull request:
## Product Description
<!-- For non-invisible changes, describe user-facing effects. -->
This PR updates the mappings which are compatible with ES 5. The first commit https://github.com/dimagi/commcare-hq/pull/33485/commits/355bf7fda181089f2c4a8bab361a63df6328a1a2 is used to transform the mappings.
The mappings are transformed as per logic - 
```
string + not_analyzed => keyword
string => text
```
The commits after that are just result of the transformation applied on each of the mapping file.

Then the next 2 commits are small updates in mappings that is done by going through the changelog.

After that the commits update the test suite to run against ES 5.

Review by commit 🐟 

## Technical Summary
<!--
    Provide a link to the ticket or document which prompted this change,
    Describe the rationale and design decisions.
-->
https://dimagi-dev.atlassian.net/browse/SAAS-14891
## Safety Assurance

### Safety story
<!--
Describe how you became confident in this change, such as
local testing, why the change is inherently safe, and/or plans to limit the blast radius of a defect.

In particular consider how existing data may be impacted by this change.
-->
Changes just affect the test suite and ensures that tests run against ES5 .
### Automated test coverage

<!-- Identify the related test coverage and the tests it would catch -->
NA
### QA Plan

<!--
- Describe QA plan that along with automated test coverages proves this PR is regression free
- Link to QA Ticket
-->
NA


### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

AmitPhulera committed:
update string to text/keyword in app mappings

AmitPhulera committed:
update string to text/keyword in case mappings

AmitPhulera committed:
update string to text/keyword in case_serach mappings

AmitPhulera committed:
update string to text/keyword in domain mappings

AmitPhulera committed:
update string to text/keyword in group mappings

AmitPhulera committed:
update string to text/keyword in sms mappings

AmitPhulera committed:
remove remove geohash, geohash_prefix, geohash_precision, lat_lon

AmitPhulera committed:
expose 9200 port of es5

AmitPhulera committed:
update warnings for elasticsearch5 urllib

AmitPhulera committed:
update docker localsettings

AmitPhulera committed:
add es5 rather that es2 as dependency

AmitPhulera committed:
add mapping transform util for converting string to text and keyword

AmitPhulera committed:
update string to text/keyword in form mappings

AmitPhulera committed:
remove null_value from text field as it is not supported and does not make sense

AmitPhulera committed:
fix tests as per new mappings

millerdev commented on docker/localsettings.py:
```diff
@@ -100,12 +100,11 @@
 
-if os.environ.get('ELASTICSEARCH_MAJOR_VERSION'):
-    ELASTICSEARCH_MAJOR_VERSION = int(os.environ.get('ELASTICSEARCH_MAJOR_VERSION'))
+ELASTICSEARCH_MAJOR_VERSION = 5
```
Can this duplicate line be removed?

AmitPhulera replied:
[945dfd8](https://github.com/dimagi/commcare-hq/pull/33485/commits/945dfd8a6dd1adcc4219e91318563c2dad305478)

millerdev reviewed (commented):
How does one decide to use `text` or `keyword` when converting from `"type": "string"`?

AmitPhulera committed:
fix remaining tests

AmitPhulera committed:
remove duplicate variable assignments

AmitPhulera committed:
lint

AmitPhulera commented:
> How does one decide to use text or keyword when converting from "type": "string"?

@millerdev I had added those details in the transform function that I created in 355bf7fda181089f2c4a8bab361a63df6328a1a2 but while cherry-picking it was missed. 
I have added it back in the PR

millerdev commented on docker/localsettings.py:
```diff
@@ -1,4 +1,4 @@
-####### Configuration for CommCareHQ Running in docker #######
+"""####### Configuration for CommCareHQ Running in docker #######"""
```
Interesting solution to that lint error. Would it also work to prefix the line with `# `?
```suggestion
# ####### Configuration for CommCareHQ Running in docker #######
```
Not saying you should change it, just trying to think of another solution.

AmitPhulera replied:
Yeah it will work with `#  `, I tried that too but did not like the extra space in the beginning 😅
